lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -27,6 +27,107 @@ using namespace mlir;
+  if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int paddings");
+  if (!matchPattern(op.dilation(), m_TorchConstantIntList(dilationInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int dilations");
+  bool ceilMode;
+  if (matchPattern(op.ceil_mode(), m_TorchConstantBool(&ceilMode))) {
+    if (ceilMode) {
+      return rewriter.notifyMatchFailure(op,
+                                         "only ceil_mode false is supported");
+    }
+  } else
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant bool ceil_mode");
+
+  return success();
+}
+
+// Computes maxpool2d for AtenMaxPool2dOp and AtenMaxPool2dWithIndicesOp.
+template <typename OpTy>
+static LogicalResult
+computeMaxPool2d(OpTy op, ConversionPatternRewriter &rewriter, Value self,
+                 SmallVector<int64_t, 2> kernelSizeInts,
+                 SmallVector<int64_t, 2> strideInts,
+                 SmallVector<int64_t, 2> paddingInts,
+                 SmallVector<int64_t, 2> dilationInts, Value &result) {
+  Location loc = op.getLoc();
+  Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+  if (!elementType.isa<mlir::FloatType>())
+    return op.emitError("unimplemented: non-floating point type");
+
+  SmallVector<int64_t, 4> paddingIncludingNC = {0, 0};
+  paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
+                            paddingInts.end());
+  Value paddedInput =
+      torch_to_linalg::getPaddedTensor(op, rewriter, self, paddingIncludingNC);
+
+  Value N = getDimOp(rewriter, loc, self, 0);
+  Value C = getDimOp(rewriter, loc, self, 1);
+  Value H = getDimOp(rewriter, loc, self, 2);
+  Value W = getDimOp(rewriter, loc, self, 3);
+
+  SmallVector<Value> paddingIntValues =
+      getAsConstantIntValues(rewriter, loc, paddingInts);
+  SmallVector<Value> dilationIntValues =
+      getAsConstantIntValues(rewriter, loc, dilationInts);
+  SmallVector<Value> kernelSizeIntValues =
+      getAsConstantIntValues(rewriter, loc, kernelSizeInts);
+  SmallVector<Value> strideIntValues =
+      getAsConstantIntValues(rewriter, loc, strideInts);
+
+  Value Hout = torch_to_linalg::getOutputDimForConvOps(
+      rewriter, loc, H, paddingIntValues[0], dilationIntValues[0],
+      kernelSizeIntValues[0], strideIntValues[0]);
+  Value Wout = torch_to_linalg::getOutputDimForConvOps(
+      rewriter, loc, W, paddingIntValues[1], dilationIntValues[1],
+      kernelSizeIntValues[1], strideIntValues[1]);
+
+  // Initialize output tensor with smallest floating point value
+  Value outTensor = rewriter.create<linalg::InitTensorOp>(
+      loc, ValueRange{N, C, Hout, Wout}, elementType);
+  auto initialAttr = rewriter.getFloatAttr(
+      elementType, APFloat::getSmallest(
```

Comment:
This isn't right. The naming of `getSmallest` is not very good at conveying its behavior, but what it does is that it returns the number of smallest *magnitude*. So in this case it returns something very close to `-0.0`. The right function is `getLargest`, which will return the largest negative number (in magnitude) possible. 

Note: I noticed that some of the tests do include negative and positive numbers; however, it seems that on average there is always a positive number around in each window to make things work out in the end. Can you add a test with only negative numbers to make sure this logic is right?